CMakeLists.txt

```diff
@@ -15,7 +15,9 @@
 # along with this program.  If not, see http://www.gnu.org/licenses.
 # -----------------------------------------------------------------------------
 
-cmake_minimum_required(VERSION 3.7)
+cmake_minimum_required(VERSION 3.7...3.22.2)
```

Comment:
Can we leave this open ended so that it continues working with future versions?
I think it's unlikely that a minor version number change will break it.

Or maybe you could make it require major version `3`, but have no requirement on the minor version.
`cmake_minimum_required(VERSION 3.7...3.*)` (i don't know cmake syntax)

Reply:
The upper end of the range does not mean that it will stop working after that...it only means that it has not been tested on a CMake version greater than that.  It affects how policy is handled. New policies introduced after the max version will warn of the new policies introduced.  I know, that seems confusing.  We could remove it if you want to keep it simpler.

Note there is a depreciation warning generated by that statement warning that this feature exists since version 3.12. By adding the max policy version I can suppress the warning for this instance.  However, it does not suppress it for all subordinate instances of CMakeLists.txt, like those in the downloaded dependencies. So I guess we just have to ignore those warnings.

Reply:
Unfortunately I cannot do the merge until it builds.

Reply:
Oh ok, if it's all just warnings then that's fine.